File: src/main/java/dev/thornium/flatcore/common/data/recipe/ComposterRecipes.java
package dev.thornium.flatcore.common.data.recipe;

import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.data.recipes.FinishedRecipe;
import net.minecraft.world.item.Item;
import net.minecraft.world.level.ItemLike;
import net.minecraft.world.level.block.ComposterBlock;

import java.util.function.Consumer;

import static dev.thornium.flatcore.gtbridge.FTRecipeTypes.COMPOSTER_RECIPES;

public class ComposterRecipes {
    private ComposterRecipes() {}

    public static void init(Consumer<FinishedRecipe> provider) {
        for (ItemLike compostable : ComposterBlock.COMPOSTABLES.keySet()) {
            Item item = compostable.asItem();

            COMPOSTER_RECIPES.recipeBuilder("compost_" + BuiltInRegistries.ITEM.getKey(item).getPath())
                    .inputItems(item)
                    .save(provider);
        }
    }
}

File: src/main/kotlin/dev/thornium/flatcore/common/data/recipe/ComposterRecipes.kt
package dev.thornium.flatcore.common.data.recipe

import dev.thornium.flatcore.gtbridge.FTRecipeTypes.COMPOSTER_RECIPES
import net.minecraft.data.recipes.FinishedRecipe
import net.minecraft.world.level.ItemLike
import net.minecraft.world.level.block.ComposterBlock
import net.minecraftforge.registries.ForgeRegistries
import java.util.function.Consumer

object ComposterRecipes {
    fun init(provider: Consumer<FinishedRecipe>) {
        for (compostable: ItemLike in ComposterBlock.COMPOSTABLES.keys) {
            val item = compostable.asItem()
            val itemId = ForgeRegistries.ITEMS.getKey(item)?.path ?: continue

            COMPOSTER_RECIPES.recipeBuilder("compost_$itemId")
                .inputItems(item)
                .save(provider)
        }
    }
}
